lib/routers/profile/index.js

```diff
@@ -20,7 +26,7 @@ const getAllProfiles = req => {
   return Promise.resolve()
     .then(() => req.user.can('profile.read.all', req.params))
     .then(allowed => {
-      if (allowed) {
+      if (allowed && !namedPeopleOnly) {
```

Comment:
I'm not sure this is enough on the basis that the alternative when this condition is not met is to show all the named people _and_ the current user.

That means that if the user looking at the list with a "all named people" selector selected will see themselves in the list, even if they are not a named person.

Reply:
includeSelf flag is now set to false when we want named people only. Self will still appear in the list if they are actually named.